ga4gh/datamodel/reads.py

```diff
@@ -553,17 +573,29 @@ def __init__(self, parentContainer, localId, readGroupHeader=None):
         self._library = None
         self._platformUnit = None
         self._runTime = None
-        if readGroupHeader is not None:
-            self._sampleId = readGroupHeader.get('SM', None)
-            self._description = readGroupHeader.get('DS', None)
-            if 'PI' in readGroupHeader:
-                self._predictedInsertSize = int(readGroupHeader['PI'])
-            self._instrumentModel = readGroupHeader.get('PL', None)
-            self._sequencingCenter = readGroupHeader.get('CN', None)
-            self._experimentDescription = readGroupHeader.get('DS', None)
-            self._library = readGroupHeader.get('LB', None)
-            self._platformUnit = readGroupHeader.get('PU', None)
-            self._runTime = readGroupHeader.get('DT', None)
+
+    def populateFromHeader(self, readGroupHeader):
+        """
+        Populate the instance variables using the specified SAM header.
+        """
+        self._sampleId = readGroupHeader.get('SM', None)
+        self._description = readGroupHeader.get('DS', None)
+        if 'PI' in readGroupHeader:
+            self._predictedInsertSize = int(readGroupHeader['PI'])
+        self._instrumentModel = readGroupHeader.get('PL', None)
+        self._sequencingCenter = readGroupHeader.get('CN', None)
+        self._experimentDescription = readGroupHeader.get('DS', None)
+        self._library = readGroupHeader.get('LB', None)
+        self._platformUnit = readGroupHeader.get('PU', None)
+        self._runTime = readGroupHeader.get('DT', None)
+
+    def populateFromRow(self, row):
+        """
+        Populate the instance variables using the specified DB row.
+        """
+        self._sampleId = row[b'sampleId']
```

Comment:
Excellent, this is exactly what is needed to be able to handle the variety of linkages that support more complicated queries.